samples/core/iris/iris.py

```diff
@@ -0,0 +1,212 @@
+  )
+
+  # Uses user-provided Python function that implements a model using TF-Learn.
+  trainer = Trainer(
+      module_file=_module_file_param,
+      custom_executor_spec=executor_spec.ExecutorClassSpec(GenericExecutor),
+      examples=transform.outputs['transformed_examples'],
+      transform_graph=transform.outputs['transform_graph'],
+      schema=infer_schema.outputs['schema'],
+      train_args=trainer_pb2.TrainArgs(num_steps=100),
+      eval_args=trainer_pb2.EvalArgs(num_steps=50)
+  )
+
+  # Get the latest blessed model for model validation.
+  model_resolver = ResolverNode(
+      instance_name='latest_blessed_model_resolver',
+      resolver_class=latest_blessed_model_resolver.LatestBlessedModelResolver,
+      model=Channel(type=Model),
+      model_blessing=Channel(type=ModelBlessing)
+  )
+
+  # Uses TFMA to compute an evaluation statistics over features of a model and
+  # perform quality validation of a candidate model (compared to a baseline).
+  # Note: to compile this successfully you'll need TFMA at >= 0.21.5
+  eval_config = tfma.EvalConfig(
+      model_specs=[
+          tfma.ModelSpec(name='candidate', label_key='variety'),
+          tfma.ModelSpec(
+              name='baseline', label_key='variety', is_baseline=True
+          )
+      ],
+      slicing_specs=[tfma.SlicingSpec()],
+      metrics_specs=[
+          tfma.MetricsSpec(
+              metrics=[
+                  tfma.MetricConfig(
+                      class_name='SparseCategoricalAccuracy',
+                      threshold=tfma.config.MetricThreshold(
+                          value_threshold=tfma.GenericValueThreshold(
+                              lower_bound={'value': 0.9}
+                          ),
+                          change_threshold=tfma.GenericChangeThreshold(
+                              direction=tfma.MetricDirection.HIGHER_IS_BETTER,
+                              absolute={'value': -1e-10}
+                          )
+                      )
+                  )
+              ]
+          )
+      ]
+  )
+
+  # Uses TFMA to compute a evaluation statistics over features of a model.
+  model_analyzer = Evaluator(
+      examples=example_gen.outputs['examples'],
+      model=trainer.outputs['model'],
+      baseline_model=model_resolver.outputs['model'],
+      # Change threshold will be ignored if there is no baseline (first run).
+      eval_config=eval_config
+  )
+
+  # Checks whether the model passed the validation steps and pushes the model
+  # to a file destination if check passed.
+  pusher = Pusher(
+      model=trainer.outputs['model'],
+      model_blessing=model_analyzer.outputs['blessing'],
+      push_destination=pusher_pb2.PushDestination(
+          filesystem=pusher_pb2.PushDestination.Filesystem(
+              base_directory=os.path.
+              join(str(pipeline.ROOT_PARAMETER), 'model_serving')
+          )
+      )
+  )
+
+  # TODO(numerology): Add pusher and model validator.
```

Comment:
refine this todo? you already have pusher.

as for model validator, remember you discussed native keras will replace it with another one? I may forget or remember it in a wrong direction.

Reply:
Are sorry I forgot to clean this up. Done.